[net7.0-rc2] Creating a cross-targeted library with net6.0-mac fails with an obscure error

<!-- 🚨 STOP 🚨 𝗦𝗧𝗢𝗣 🚨 𝑺𝑻𝑶𝑷 🚨 -->

<!--
Please fill in the *entire* template below.
The template is designed to avoid unnecessary delays to confirm and fix reported issues. Issues with missing information generally end up with a `need-info` label and take significantly more time to fix. Please help us help you!
For more information see https://github.com/xamarin/xamarin-macios/wiki/Submitting-Bugs-&-Suggestions
For support requests use https://www.xamarin.com/support
-->
### Steps to Reproduce
Without having the macos SDK installed, build the following:
```xml
<Project Sdk="Microsoft.NET.Sdk">
	<PropertyGroup>
		<TargetFrameworks>net6.0-macos</TargetFrameworks>
	</PropertyGroup>
</Project>
```

### Expected Behavior

An error from .NET SDK to flag that a workload is missing.

### Actual Behavior

```
1>C:\Program Files\dotnet\sdk\7.0.100-rc.2.22477.23\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.targets(1220,3): 
error MSB4019: The imported project "C:\Program Files\dotnet\sdk\7.0.100-rc.2.22477.23\Sdks\Microsoft.NET.Sdk\12.3.1186-rc.2\targets\Xamarin.Shared.Sdk.MultiTarget.targets" was not found. 
Confirm that the expression in the Import declaration ";..\12.3.1186-rc.2\targets\Xamarin.Shared.Sdk.MultiTarget.targets" is correct, and that the file exists on disk.
```

### Environment

```
dotnet --version
7.0.100-rc.2.22477.23
```
```
dotnet workload list

Installed Workload Id      Manifest Version                         Installation Source
---------------------------------------------------------------------------------------
ios                        16.0.1271-rc.2-xcode14/7.0.100-rc.2      VS 17.4.33015.44
maui-android               7.0.0-rc.2.6866/7.0.100-rc.2             VS 17.4.33015.44
maui-windows               7.0.0-rc.2.6866/7.0.100-rc.2             VS 17.4.33015.44
maui-maccatalyst           7.0.0-rc.2.6866/7.0.100-rc.2             VS 17.4.33015.44
android                    33.0.0-rc.2.202/7.0.100-rc.2             VS 17.4.33015.44
maccatalyst                15.4.1186-rc.2/7.0.100-rc.2              VS 17.4.33015.44
maui-ios                   7.0.0-rc.2.6866/7.0.100-rc.2             VS 17.4.33015.44

Use `dotnet workload search` to find additional workloads to install.
```

### Build Logs

<!--
1. How to get a build log: https://github.com/xamarin/xamarin-macios/wiki/Diagnosis#build-logs
2. Place the cursor below this comment block.
3. Drag and drop the build log.
-->

I am hitting this error message, even though the macos workload is installed:

```
Installierte Workload-ID      Manifestversion        Installationsquelle
-----------------------------------------------------------------------------------
android                       33.0.4/7.0.100         SDK 7.0.100, VS 17.4.33213.308
ios                           16.1.1477/7.0.100      SDK 7.0.100, VS 17.4.33213.308
maccatalyst                   16.1.1477/7.0.100      SDK 7.0.100, VS 17.4.33213.308
macos                         13.0.2032/7.0.100      SDK 7.0.100
maui-windows                  7.0.52/7.0.100         VS 17.4.33213.308
maui-maccatalyst              7.0.52/7.0.100         VS 17.4.33213.308
maui-ios                      7.0.52/7.0.100         VS 17.4.33213.308
maui-android                  7.0.52/7.0.100         VS 17.4.33213.308
```

Build error:

`1>C:\Program Files\dotnet\sdk\7.0.102\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.targets(1226,3): error MSB4019: The imported project "C:\Program Files\dotnet\sdk\7.0.102\Sdks\Microsoft.NET.Sdk\13.0.2032\targets\Xamarin.Shared.Sdk.MultiTarget.targets" was not found. Confirm that the expression in the Import declaration ";..\13.0.2032\targets\Xamarin.Shared.Sdk.MultiTarget.targets" is correct, and that the file exists on disk.
`

I am trying to create a new cross-platform UNO library, but hitting this bug. Any ideas? How can I work around this?

--- a/dotnet/Workloads/WorkloadManifest.MacCatalyst.template.targets
+++ b/dotnet/Workloads/WorkloadManifest.MacCatalyst.template.targets
@@ -9,6 +9,7 @@
 	</ItemGroup>
 
 	<PropertyGroup Condition=" '$(TargetPlatformIdentifier)' == 'MacCatalyst' and $([MSBuild]::VersionEquals($(TargetFrameworkVersion), '6.0')) ">
-		<AfterMicrosoftNETSdkTargets>$(AfterMicrosoftNETSdkTargets);$(_XamarinSdkRootDirectory)..\@NUGET_VERSION_NO_METADATA@\targets\Xamarin.Shared.Sdk.MultiTarget.targets</AfterMicrosoftNETSdkTargets>
+		<_MultiTargetTargetsFile>$(_XamarinSdkRootDirectory)..\@NUGET_VERSION_NO_METADATA@\targets\Xamarin.Shared.Sdk.MultiTarget.targets</_MultiTargetTargetsFile>
+		<AfterMicrosoftNETSdkTargets Condition="Exists('$(_MultiTargetTargetsFile)')">$(AfterMicrosoftNETSdkTargets);$(_MultiTargetTargetsFile)</AfterMicrosoftNETSdkTargets>
 	</PropertyGroup>
 </Project>

--- a/dotnet/Workloads/WorkloadManifest.iOS.template.targets
+++ b/dotnet/Workloads/WorkloadManifest.iOS.template.targets
@@ -11,6 +11,7 @@
 	</ItemGroup>
 
 	<PropertyGroup Condition=" '$(TargetPlatformIdentifier)' == 'iOS' and $([MSBuild]::VersionEquals($(TargetFrameworkVersion), '6.0')) ">
-		<AfterMicrosoftNETSdkTargets>$(AfterMicrosoftNETSdkTargets);$(_XamarinSdkRootDirectory)..\@NUGET_VERSION_NO_METADATA@\targets\Xamarin.Shared.Sdk.MultiTarget.targets</AfterMicrosoftNETSdkTargets>
+		<_MultiTargetTargetsFile>$(_XamarinSdkRootDirectory)..\@NUGET_VERSION_NO_METADATA@\targets\Xamarin.Shared.Sdk.MultiTarget.targets</_MultiTargetTargetsFile>
+		<AfterMicrosoftNETSdkTargets Condition="Exists('$(_MultiTargetTargetsFile)')">$(AfterMicrosoftNETSdkTargets);$(_MultiTargetTargetsFile)</AfterMicrosoftNETSdkTargets>
 	</PropertyGroup>
 </Project>

--- a/dotnet/Workloads/WorkloadManifest.macOS.template.targets
+++ b/dotnet/Workloads/WorkloadManifest.macOS.template.targets
@@ -9,6 +9,7 @@
 	</ItemGroup>
 
 	<PropertyGroup Condition=" '$(TargetPlatformIdentifier)' == 'macOS' and $([MSBuild]::VersionEquals($(TargetFrameworkVersion), '6.0')) ">
-		<AfterMicrosoftNETSdkTargets>$(AfterMicrosoftNETSdkTargets);$(_XamarinSdkRootDirectory)..\@NUGET_VERSION_NO_METADATA@\targets\Xamarin.Shared.Sdk.MultiTarget.targets</AfterMicrosoftNETSdkTargets>
+		<_MultiTargetTargetsFile>$(_XamarinSdkRootDirectory)..\@NUGET_VERSION_NO_METADATA@\targets\Xamarin.Shared.Sdk.MultiTarget.targets</_MultiTargetTargetsFile>
+		<AfterMicrosoftNETSdkTargets Condition="Exists('$(_MultiTargetTargetsFile)')">$(AfterMicrosoftNETSdkTargets);$(_MultiTargetTargetsFile)</AfterMicrosoftNETSdkTargets>
 	</PropertyGroup>
 </Project>

--- a/dotnet/Workloads/WorkloadManifest.tvOS.template.targets
+++ b/dotnet/Workloads/WorkloadManifest.tvOS.template.targets
@@ -9,6 +9,7 @@
 	</ItemGroup>
 
 	<PropertyGroup Condition=" '$(TargetPlatformIdentifier)' == 'tvOS' and $([MSBuild]::VersionEquals($(TargetFrameworkVersion), '6.0')) ">
-		<AfterMicrosoftNETSdkTargets>$(AfterMicrosoftNETSdkTargets);$(_XamarinSdkRootDirectory)..\@NUGET_VERSION_NO_METADATA@\targets\Xamarin.Shared.Sdk.MultiTarget.targets</AfterMicrosoftNETSdkTargets>
+		<_MultiTargetTargetsFile>$(_XamarinSdkRootDirectory)..\@NUGET_VERSION_NO_METADATA@\targets\Xamarin.Shared.Sdk.MultiTarget.targets</_MultiTargetTargetsFile>
+		<AfterMicrosoftNETSdkTargets Condition="Exists('$(_MultiTargetTargetsFile)')">$(AfterMicrosoftNETSdkTargets);$(_MultiTargetTargetsFile)</AfterMicrosoftNETSdkTargets>
 	</PropertyGroup>
 </Project>

